gprofiler/metadata/system_metadata.py

```diff
@@ -0,0 +1,219 @@
+        return "k8s", "k8s"
+    elif os.getenv("GPROFILER_IN_CONTAINER") is not None:  # set by our Dockerfile
+        return "container", "containers"
+    elif os.getenv("STATICX_BUNDLE_DIR") is not None:  # set by staticx
+        return "standalone_executable", "instances"
+    else:
+        return "local_python", "instances"
+
+
+def get_local_ip() -> str:
+    try:
+        local_ips = [ip for ip in socket.gethostbyname_ex(socket.gethostname())[2] if not ip.startswith("127.")]
+    except Exception:
+        local_ips = []
+    if not local_ips:
+        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        try:
+            s.connect(("8.8.8.8", 53))
+            local_ips.append(s.getsockname()[0])
+        finally:
+            s.close()
+    return local_ips[0] if local_ips else "unknown"
+
+
+def get_mac_address() -> str:
+    # This way is used to ensure consistency
+
+    IS_64BIT = sys.maxsize > 2 ** 32
+    IFNAMSIZ = 16
+    IFF_LOOPBACK = 8
+    MAC_BYTES_LEN = 6
+    SIZE_OF_SHORT = struct.calcsize('H')
+    SIZE_OF_STUCT_ifreq = 40 if IS_64BIT else 32
+    MAX_BYTE_COUNT = 1024
+
+    buf = array.array('B', b'\0' * MAX_BYTE_COUNT)
+    ifc = struct.pack('iL', MAX_BYTE_COUNT, buf.buffer_info()[0])
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_IP)
+    outbytes = struct.unpack('iL', fcntl.ioctl(s.fileno(), 0x8912, ifc))[0]  # SIOCGIFCONF
+    data = buf.tobytes()[:outbytes]
+    for index in range(0, len(data), SIZE_OF_STUCT_ifreq):
+        iface = data[index : index + SIZE_OF_STUCT_ifreq]
+        res = fcntl.ioctl(s.fileno(), 0x8913, iface)  # SIOCGIFFLAGS
+        ifr_flags = struct.unpack(f'{IFNAMSIZ}sH', res[: IFNAMSIZ + SIZE_OF_SHORT])[1]
+        is_loopback = ifr_flags & IFF_LOOPBACK
+        if is_loopback:
+            continue
+        res = fcntl.ioctl(s.fileno(), 0x8927, iface)  # SIOCGIFHWADDR
+        address = struct.unpack(f'{IFNAMSIZ}sH{MAC_BYTES_LEN}s', res[: IFNAMSIZ + SIZE_OF_SHORT + MAC_BYTES_LEN])[2]
+        mac = struct.unpack(f'{MAC_BYTES_LEN}B', address)
+        address = ":".join(['%02X' % i for i in mac])
+        return address
+    return "unknown"
+
+
+@dataclass
+class SystemInfo:
+    python_version: str
+    run_mode: str
+    deployment_type: str
+    kernel_release: str
+    kernel_version: str
+    system_name: str
+    processors: int
+    memory_capacity_mb: int
+    hostname: str
+    os_name: str
+    os_release: str
+    os_codename: str
+    libc_type: str
+    libc_version: str
+    hardware_type: str
+    pid: int
+    mac_address: str
+    private_ip: str
+    spawn_uptime_ms: float
+
+
+def get_static_system_info() -> SystemInfo:
+    hostname, distribution, libc_tuple, mac_address, local_ip = _initialize_system_info()
+    clock_boottime_type = getattr(time, "CLOCK_BOOTTIME", None)
+    assert clock_boottime_type is not None, "The gProfiler only supports Unix with a kernel version of >=2.6.39"
```

Comment:
We support CentOS 6 (I tested on 2.6.32).

So don't remove that support - instead, use another clock or so. (although specifically in the CentOS case, I'm fairly certain they have backported `CLOCK_BOOTTIME`)